### ISSUE TITLE
fh / mymaster with several bugs in the last builds

### ISSUE THREAD

**Issue by spielvan** [reicast/reicast-emulator#1505]
From build 3320669 of 13.02.2019 that started these bugs making it impossible to use the new builds, I will try to list the bugs that I remember head and that before did not have:

1. When you click on about to know the build version, instead of loading the list of builds, simply close the reicast.

2. With the Xbox 360 game pad, by the x-input, the new menu gui, does not recognize the gamepad, inpossible to use, in previous versions it was compatible in the new GUI menu, the mouse still works.

3. When I exit the game to the reicast menu, instead of returning to the menu, it closes the reicast.

4. Sometimes when you are going to load a game it closes the reicast.

**Comment by flyinghead** [comment]
Thanks for this detailed report.

Regarding issue 4 above, try to use the Send Logs option when you restart reicast. It might contain useful information about the crash.

**Comment by flyinghead** [comment]
For the Xbox 360 gamepad problem, can you check for a mapping file in the `[config]/mappings` directory? Try to delete it if it exists.
Also can you remap the gamepad buttons?

**Comment by spielvan** [comment]
I tried to delete the mapper and did the remapping through the new menu, but it does not reconfigure the game pad in the new gui menu, just the rest of the reicast and the game.

**Comment by hemix68** [comment]
Hi. I have a Xiaomi mi mix 2s with Eu beta 9.2.21.
In build r8. 1-240-g2b3a052a[498692236], i have crash if open option.
If open a game, i have a screen black.

**Comment by flyinghead** [comment]
@hemix68 This is not the same branch so your problem probably as nothing to do with these issues.

Please open a new issue.

**Comment by flyinghead** [comment]
I just rechecked on Windows and I have no issue using a simulated xbox 360 gamepad.

Note that only the first xinput controller can control the menu. The remaining three cannot.

Also use these builds to have the latest features and fixes: http://builds.reicast.com/#fh/content-browser

**Comment by spielvan** [comment]
> I just rechecked on Windows and I have no issue using a simulated xbox 360 gamepad.
> 
> Note that only the first xinput controller can control the menu. The remaining three cannot.
> 
> Also use these builds to have the latest features and fixes: http://builds.reicast.com/#fh/content-browser

I started using the content-browser version and actually managed to run the game pad of xbox 360 with it, there is only one problem in this part, when I exit and close the reicast and enter it again by default the control of xbox 360 comes as being control B, then you have to go again with the mouse and put to control A to run the game pad, it does not save in A when it closes reicast, but I think this bug should be something simple to solve and keep A by default , since all other game pad types are in A by default.

**Comment by flyinghead** [comment]
On Windows, the first Xinput controller is connected to port A, the second to port B and so on. So for some reason your xbox 360 controller is the second Xinput device.

Do you have something showing on port A besides the keyboard and mouse?

**Comment by spielvan** [comment]
> On Windows, the first Xinput controller is connected to port A, the second to port B and so on. So for some reason your xbox 360 controller is the second Xinput device.
> 
> Do you have something showing on port A besides the keyboard and mouse?

I'm using a ps2 control adapter for xbox 360, but it shows from A to D, and this with the port LED on, I'm running reicast on Android.

**Comment by flyinghead** [comment]
I don't know why I keep thinking that you're on Windows. Anyway...

On Android, if a touch screen is present, it is used for the Virtual Gamepad on port A. Then the first external gamepad is on Port B.

Right now the port setting is not saved.

**Comment by spielvan** [comment]
> I don't know why I keep thinking that you're on Windows. Anyway...
> 
> On Android, if a touch screen is present, it is used for the Virtual Gamepad on port A. Then the first external gamepad is on Port B.
> 
> Right now the port setting is not saved.

So this is the reason why I change to port A, whenever I open it will come as port B. But rest of the way I like this very stable build, I'll continue with the tests on it, thank you very much.

**Comment by spielvan** [comment]
@flyinghead my dear reminds you of the bug that of the gamepad by x-input did not catch first on the new menu gui of reicast, there had to go with the mouse in the settings and change the x-input of gamepad B to gamepad A, with the build master 4dca8a2 I managed to solve it, you click on the reicast icon, open it and press down on the dpad of the game pad once and it will leave the Settings options light blue and you can use the x-input gamepad normally without needing mouse. A hug and thank you for everything from the heart.

**Comment by flyinghead** [comment]
yes, the maple port should now be saved starting with 66580b16c6e32b337a19312db03d93a1fa99c76c
